core/src/main/java/org/bitcoinj/store/LevelDbFullPrunedBlockStore.java

```diff
@@ -0,0 +1,1070 @@
+package org.bitcoinj.store;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.io.*;
+import java.nio.ByteBuffer;
+
+import org.bitcoinj.core.Address;
+import org.bitcoinj.core.AddressFormatException;
+import org.bitcoinj.core.NetworkParameters;
+import org.bitcoinj.core.ScriptException;
+import org.bitcoinj.core.Sha256Hash;
+import org.bitcoinj.core.StoredBlock;
+import org.bitcoinj.core.StoredUndoableBlock;
+import org.bitcoinj.core.Transaction;
+import org.bitcoinj.core.TransactionOutputChanges;
+import org.bitcoinj.core.UTXO;
+import org.bitcoinj.core.UTXOProviderException;
+import org.bitcoinj.core.VerificationException;
+import org.bitcoinj.script.Script;
+import org.iq80.leveldb.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.iq80.leveldb.impl.Iq80DBFactory.*;
+
+import com.google.common.collect.Lists;
+
+/**
+ * 
+ * <p>An implementation of a Fully Pruned Block Store using a Pure Java leveldb
+ * implementation as the backing data store.<p>
+ * 
+ * <p>Includes number of caches to optimise the initial blockchain download</p> 
+ *  
+ *
+ */
+
+
+public class LevelDbFullPrunedBlockStore implements FullPrunedBlockStore {
+	private static final Logger log = LoggerFactory.getLogger(LevelDbFullPrunedBlockStore.class);

```

Comment:
Leading tab/space mismatches all over the place. We're using 4 spaces per indent level.